src/frontends/tensorflow/src/input_model.cpp

```diff
@@ -171,6 +171,7 @@ std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::get_op_place
 std::vector<std::shared_ptr<OpPlace>> InputModel::InputModelTFImpl::determine_cut_nodes() const {
     std::vector<std::shared_ptr<OpPlace>> topologically_sorted_ops;
     std::stack<std::shared_ptr<OpPlace>> ops_to_do;
+    std::unordered_set<std::string> ops_names_to_do;
```

Comment:
why do we need `ops_names_to_do`? only to check that NextIteration is not in the queue?
maybe it's better to store pointers to OpPlace (OpPlace*) in this set instead of names?

Reply:
yes, it is to check that the same node is added to the deque only once.

Reply:
Fixed.